src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -28,26 +32,31 @@ internal sealed class LoadedProject : IDisposable
     /// The most recent version of the project design time build information; held onto so the next reload we can diff against this.
     /// </summary>
     private ProjectFileInfo? _mostRecentFileInfo;
+    /// <summary>
+    /// The most recent version of the file glob matcher.  Held onto 
+    /// </summary>
+    private Lazy<ImmutableArray<Matcher>>? _mostRecentFileMatchers;
     private IWatchedFile? _mostRecentProjectAssetsFileWatcher;
     private ImmutableArray<CommandLineReference> _mostRecentMetadataReferences = ImmutableArray<CommandLineReference>.Empty;
     private ImmutableArray<CommandLineAnalyzerReference> _mostRecentAnalyzerReferences = ImmutableArray<CommandLineAnalyzerReference>.Empty;
 
     public LoadedProject(ProjectSystemProject projectSystemProject, SolutionServices solutionServices, IFileChangeWatcher fileWatcher, ProjectTargetFrameworkManager targetFrameworkManager)
     {
         Contract.ThrowIfNull(projectSystemProject.FilePath);
+        _projectFilePath = projectSystemProject.FilePath;
 
         _projectSystemProject = projectSystemProject;
         _optionsProcessor = new ProjectSystemProjectOptionsProcessor(projectSystemProject, solutionServices);
         _targetFrameworkManager = targetFrameworkManager;
 
         // We'll watch the directory for all source file changes
         // TODO: we only should listen for add/removals here, but we can't specify such a filter now
-        var projectDirectory = Path.GetDirectoryName(projectSystemProject.FilePath)!;
+        _projectDirectory = Path.GetDirectoryName(_projectFilePath)!;
 
         _fileChangeContext = fileWatcher.CreateContext([
-            new(projectDirectory, ".cs"),
-            new(projectDirectory, ".cshtml"),
-            new(projectDirectory, ".razor")
+            new(_projectDirectory, ".cs"),
```

Comment:
Are we not watching for changes in non-Razor additional files, which might affect other source generators? How about resx files, etc.?

Reply:
We are watching for changes for files we already know about in the solution elsewhere.  This is attempting to look for added/removed files that we don't know about that we might need to reload the project for.

Yes - this file watcher has problems.  Additions of files outside the project cone are not watched.  Additions of other kinds of files are also not watched.   However that is pre-existing and not directly the problem I'm trying to solve at the moment.

Reply:
Note - this code only applies to C# standalone (without Dev Kit).